File: 58. Length of Last Word.java
class Solution {
    public int lengthOfLastWord(String s) {
        if (s == null || s == "") {
            return 0;
        }

        int lastWordLen = 0;
        boolean isLastWord = false;
        for (int i = s.length() - 1; i >= 0; --i) {
            if (s.charAt(i) == ' ' && isLastWord) {
                return lastWordLen;
            }

            if (s.charAt(i) != ' ') {
                ++lastWordLen;
                isLastWord = true;
            }
        }

        return lastWordLen;
    }

    public int lengthOfLastWord1(String s) {
        if (s == null || s == "") {
            return 0;
        }

        int idx = s.lastIndexOf(" ");
        if (idx < 0) {
            return s.length();
        }

        while (idx > 0 && idx == s.length() - 1) {
            s = s.substring(0, idx);
            idx = s.lastIndexOf(" ");
        }

        return (s.substring(idx + 1)).length();
    }
}

File: 58. Length of Last Word.kt
class Solution {
    fun lengthOfLastWord(s: String): Int {
        var wordList = s.trim().split(' ')

        return wordList.get(wordList.size - 1).length
    }


    fun lengthOfLastWord1(s: String): Int {
        var lastWordLen = 0

        for (i in s.length-1 downTo 0) {
            if (s[i] == ' ') {
                if (lastWordLen == 0) {
                    continue
                } else {
                    return lastWordLen
                }
            } else {
                ++lastWordLen
            }
        }

        return lastWordLen
    }
}
